crates/ruff_formatter/src/format_element/document.rs

```diff
@@ -255,11 +255,35 @@ impl Format<IrFormatContext<'_>> for &[FormatElement] {
 
                     in_text = true;
 
+                    fn write_escaped(
+                        element: &FormatElement,
+                        f: &mut Formatter<IrFormatContext>,
+                    ) -> FormatResult<()> {
+                        let text = match element {
+                            FormatElement::StaticText { text } => text,
+                            FormatElement::DynamicText { text } => text.as_ref(),
+                            FormatElement::SourceCodeSlice { slice, .. } => {
+                                slice.text(f.context().source_code())
+                            }
+                            _ => unreachable!(),
+                        };
+
+                        if text.contains('"') {
+                            f.write_element(FormatElement::DynamicText {
+                                text: text.replace('"', "\\\"").into(),
```

Comment:
Maybe that's a bit easier to read

```suggestion
                                text: text.replace('"', r#"\""#).into(),
```